### PR TITLE
LPS-165648 Create simple repository browser tag

### PR DESCRIPTION
**Pull request by adolfopa**
# What is this about?

This is a tag that offers a limited interface to browse and operate on
repository documents and folders.

image

# How do I test this?

You'll need to add the tag to a portlet JSP (ideally, one with no other content):
```
<liferay-document-library:repository-browser />
```

You can specify a particular repository to browse with the `repositoryId` attribute:
```
<liferay-document-library:repository-browser repositoryId="<%= repositoryId %>" />
```

After that, you should be able to:

1. Navigate through the folder structure,
2. Delete documents, shortcuts and folders,
3. Rename documents, folders and shortcuts,
4. Upload a single document,
5. Create a folder,
6. Search entries by name.

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:relevant

File: modules/apps/document-library/document-library-taglib/src/main/java/com/liferay/document/library/taglib/servlet/taglib/RepositoryBrowserTag.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	}
+
+	@Override
+	public void setPageContext(PageContext pageContext) {
+		super.setPageContext(pageContext);
+
+		setServletContext(ServletContextUtil.getServletContext());
+	}
+
+	public void setRepositoryId(long repositoryId) {
```
Comment (AliciaGarciaGarcia) [inline modules/apps/document-library/document-library-taglib/src/main/java/com/liferay/document/library/taglib/servlet/taglib/RepositoryBrowserTag.java:54]: Is this going to be used at some point?

**Review by AliciaGarciaGarcia — APPROVED** [review]

File: modules/apps/document-library/document-library-taglib/src/main/java/com/liferay/document/library/taglib/servlet/taglib/RepositoryBrowserTag.java (hunk 1 above)
Comment (adolfopa) [reply modules/apps/document-library/document-library-taglib/src/main/java/com/liferay/document/library/taglib/servlet/taglib/RepositoryBrowserTag.java:54]: Yes. That will be used most of the times; the intended use case of this tag is to display private repositories (and in that case, you need to provide a repository id).

**Review by AliciaGarciaGarcia — APPROVED** [review]

**Review by marcogalluzzi — APPROVED** [review]
Looks good to me

File: modules/apps/document-library/document-library-taglib/src/main/resources/META-INF/resources/repository_browser/js/RepositoryBrowserManagementToolbarPropsTransformer.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+					formSubmitURL: createFolderURL,
+					mainFieldLabel: Liferay.Language.get('name'),
+					mainFieldName: 'name',
+					method: 'PUT',
+					namespace: '',
+					onFormSuccess: () => window.location.reload(),
+				});
+			}
+			else if (item?.data?.action === 'uploadFile') {
+				window[`${portletNamespace}UploadFile`]();
```
[the remark this answers is not in the document]
Comment (ambrinchaudhary) [reply modules/apps/document-library/document-library-taglib/src/main/resources/META-INF/resources/repository_browser/js/RepositoryBrowserManagementToolbarPropsTransformer.js:79]: I guess you are right, let me test it and if it works I'll add a commit with this change

**Review by ambrinchaudhary — APPROVED** [review]
LGTM!

**Comment by ambrinchaudhary** [comment]
ci:test:sf

**Comment by ambrinchaudhary** [comment]
ci:test:relevant

**Comment by ambrinchaudhary** [comment]
ci:forward

**Comment by Hanlf** [comment]
ci:forward

**Comment by Hanlf** [comment]
Hi @adolfopa 
The [modules-semantic-versioning-jdk8/0/0](https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/100316//consoleText) error should be related to this PR, could you take a look, thanks!
```
     [exec] > Task :apps:document-library:document-library-taglib:baseline
     [exec]   PACKAGE_NAME                                       DELTA      CUR_VER    BASE_VER   REC_VER    WARNINGS  
     [exec] = ================================================== ========== ========== ========== ========== ==========
     [exec] * com.liferay.document.library.taglib.servlet.taglib MINOR      1.0.0      1.0.0      1.1.0      VERSION INCREASE REQUIRED
     [exec] 	+   class      com.liferay.document.library.taglib.servlet.taglib.RepositoryBrowserTag
     [exec] 		+   extends    com.liferay.taglib.util.IncludeTag
     [exec] 		+   method     <init>()
     [exec] 			+   return     void
     [exec] 		+   method     clone()
     [exec] 			+   access     protected
     [exec] 			+   return     java.lang.Object
     [exec] 		+   method     doStartTag()
     [exec] 			+   return     int
     [exec] 		+   method     equals(java.lang.Object)
     [exec] 			+   return     boolean
     [exec] 		+   method     finalize()
     [exec] 			+   access     protected
     [exec] 			+   return     void
     [exec] 		+   method     getPage()
     [exec] 			+   access     protected
     [exec] 			+   return     java.lang.String
     [exec] 		+   method     getRepositoryId()
     [exec] 			+   return     long
     [exec] 		+   method     hashCode()
     [exec] 			+   return     int
     [exec] 		+   method     setAttributes(javax.servlet.http.HttpServletRequest)
     [exec] 			+   access     protected
     [exec] 			+   return     void
     [exec] 		+   method     setPageContext(javax.servlet.jsp.PageContext)
     [exec] 			+   return     void
     [exec] 		+   method     setRepositoryId(long)
     [exec] 			+   return     void
     [exec] 		+   method     toString()
     [exec] 			+   return     java.lang.String
     [exec] [Baseline Warning] Bundle Version Change Recommended: 3.1.0
     [exec] [Baseline Report] Mode: diff (persisted)
     [exec] Semantic versioning is incorrect while checking /opt/dev/projects/github/liferay-portal/tools/sdk/dist/com.liferay.document.library.taglib-3.0.9.jar against /opt/dev/projects/github/liferay-portal/.gradle/caches/modules-2/files-2.1/com.liferay/com.liferay.document.library.taglib/3.0.8/5ef7a2a74eeb71102ceae9385213e1c04479c5e0/com.liferay.document.library.taglib-3.0.8.jar
     [exec] 
```

**Comment by adolfopa** [comment]
Ran baseline, rebased and push-force'd; no other changes.

**Comment by adolfopa** [comment]
ci:forward

**Pull request closed without merging**